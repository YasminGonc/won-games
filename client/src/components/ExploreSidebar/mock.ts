import { ExploreSideBarProps } from '.'

export const exploreSideBarMock: Pick<ExploreSideBarProps, 'items'> = {
  items: [
    {
      title: 'Price',
      type: 'checkbox',
      fields: [
        'Under $50',
        'Under $100',
        'Under $150',
        'Under $200',
        'Free',
        'Discouted'
      ]
    },
    {
      title: 'Sort by',
      name: 'sort_by',
      type: 'radio',
      fields: ['High to low', 'Low to high']
    },
    {
      title: 'System',
      type: 'checkbox',
      fields: ['Windows', 'Linux', 'MAC']
    },
    {
      title: 'Genre',
      type: 'checkbox',
      fields: ['Action', 'Adventure', 'Simulation']
    }
  ]
}
