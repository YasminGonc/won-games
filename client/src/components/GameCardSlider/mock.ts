import { GameCardProps } from '../GameCard'

const GameCardSliderMock: GameCardProps[] = [
  {
    title: 'Population Zero 1',
    developer: 'Rockstar Games',
    img: 'https://source.unsplash.com/user/willianjusten/300x140',
    price: '$ 215,00',
    promotionPrice: '$ 100,00',
    favorite: false,
    hasRibbon: true,
    ribbon: '20% OFF',
    ribbonSize: 'small',
    ribbonColor: 'primary'
  },
  {
    title: 'Population Zero 2',
    developer: 'Rockstar Games',
    img: 'https://source.unsplash.com/user/willianjusten/300x141',
    price: '$ 215,00'
  },
  {
    title: 'Population Zero 3',
    developer: 'Rockstar Games',
    img: 'https://source.unsplash.com/user/willianjusten/300x142',
    price: '$ 215,00'
  },
  {
    title: 'Population Zero 4',
    developer: 'Rockstar Games',
    img: 'https://source.unsplash.com/user/willianjusten/300x143',
    price: '$ 215,00'
  },
  {
    title: 'Population Zero 5',
    developer: 'Rockstar Games',
    img: 'https://source.unsplash.com/user/willianjusten/300x144',
    price: '$ 215,00'
  },
  {
    title: 'Population Zero 6',
    developer: 'Rockstar Games',
    img: 'https://source.unsplash.com/user/willianjusten/300x145',
    price: '$ 215,00'
  }
]

export default GameCardSliderMock
