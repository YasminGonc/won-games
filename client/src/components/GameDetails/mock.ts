import { GameDetailsProps } from '.'

export const GameDetailsMock: GameDetailsProps = {
  developer: 'Gearbox Software',
  publisher: '2K',
  platforms: ['linux', 'windowns', 'mac'],
  realeaseDate: '2020-11-21T23:00:00',
  rating: 'BR0',
  genres: ['Role-playing', 'Narrative']
}
