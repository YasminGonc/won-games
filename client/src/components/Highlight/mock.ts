import { HighlightProps } from '.'

const HighlighMock: HighlightProps = {
  title: 'Read Dead is back',
  subtitle: "Come see John's new adventures",
  backgroundImage: '/img/red-dead-img.png',
  buttonLabel: 'Buy now',
  buttonLink: '/rdr2',
  alignment: 'right'
}

export default HighlighMock
