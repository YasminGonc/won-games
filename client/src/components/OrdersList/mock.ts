import { OrdersListProps } from '.'

export const ordersListMock: OrdersListProps = {
  items: [
    {
      img: 'https://source.unsplash.com/user/willianjusten/151x70',
      title: 'Red Dead Redemption 2',
      price: '$ 215,00',
      downloadLink: 'https://wongames.com/game/download/1',
      paymentInfo: {
        flag: 'mastercard',
        img: '/img/mastercard.png',
        number: '**** **** **** 4326',
        purchaseDate: 'Purchase made on 07/20/2022 at 20:23'
      }
    },
    {
      img: 'https://source.unsplash.com/user/willianjusten/151x71',
      title: 'Red Dead Redemption 2',
      price: '$ 215,00',
      downloadLink: 'https://wongames.com/game/download/2',
      paymentInfo: {
        flag: 'visa',
        img: '/img/visa.png',
        number: '**** **** **** 5423',
        purchaseDate: 'Purchase made on 07/20/2022 at 20:00'
      }
    }
  ]
}
