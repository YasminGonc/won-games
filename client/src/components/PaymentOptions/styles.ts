import styled, { css } from 'styled-components'
import media from 'styled-media-query'
import * as RadioGroup from '@radix-ui/react-radio-group'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    background: ${theme.colors.white};
  `}
`

export const Body = styled.div`
  ${({ theme }) => css`
    padding: ${theme.spacings.xsmall} ${theme.spacings.small} 0
      ${theme.spacings.small};
  `}
`

export const CardItem = styled.div``

export const RadioGroupRoot = styled(RadioGroup.Root)`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    gap: ${theme.spacings.xsmall};
  `}
`

export const ItemWrapper = styled.div`
  ${({ theme }) => css`
    background-color: ${theme.colors.lightGray};
    padding: ${theme.spacings.xxsmall};

    display: flex;
    align-items: center;
    justify-content: space-between;
  `}
`

export const Label = styled.label`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    gap: ${theme.spacings.xsmall};
    cursor: pointer;
  `}
`

export const RadioGroupItem = styled(RadioGroup.Item)`
  ${({ theme }) => css`
    background-color: transparent;
    border-radius: 100%;
    border: 2px solid ${theme.colors.darkGray};

    width: 21px;
    height: 21px;
    cursor: pointer;

    &:focus {
      box-shadow: 0 0 0.5rem ${theme.colors.darkGray};
    }

    &[data-state='checked'] {
      border-color: ${theme.colors.primary};
    }
  `}
`

export const RadioGroupIndicator = styled(RadioGroup.Indicator)`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: center;

    width: 100%;
    height: 100%;
    position: relative;

    &::after {
      content: '';
      display: block;
      width: 11px;
      height: 11px;
      border-radius: 50%;
      background-color: ${theme.colors.primary};
    }
  `}
`

export const AddCard = styled(ItemWrapper)`
  ${({ theme }) => css`
    justify-content: left;
    margin-top: ${theme.spacings.xsmall};
    cursor: pointer;

    svg {
      margin-right: ${theme.spacings.xsmall};
    }
  `}
`

export const Footer = styled.div`
  ${({ theme }) => css`
    background-color: ${theme.colors.lightGray};
    margin-top: ${theme.spacings.large};
    padding: ${theme.spacings.xsmall} ${theme.spacings.small};

    display: flex;
    flex-direction: column;
    gap: ${theme.spacings.xsmall};

    ${media.greaterThan('medium')`
      flex-direction: row;
    `}
  `}
`
