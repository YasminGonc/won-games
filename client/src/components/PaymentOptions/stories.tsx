import { StoryObj, Meta } from '@storybook/react'
import { PaymentOption, PaymentOptionProps } from '.'

import { paymentOptionsMock } from './mock'

export default {
  title: 'PaymentOption',
  component: PaymentOption,
  args: {
    cards: paymentOptionsMock
  }
} as Meta<PaymentOptionProps>

export const Default: StoryObj<PaymentOptionProps> = {}
