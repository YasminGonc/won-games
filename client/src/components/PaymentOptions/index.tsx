import * as S from './styles'
import { Plus, ShoppingCart } from 'phosphor-react'
import { Heading } from '../Heading'
import Button from '../Button'

export type PaymentCard = {
  number: string
  flag: string
  img: string
}

export type PaymentOptionProps = {
  cards?: PaymentCard[]
  handlePayment?: () => void
}

export function PaymentOption({ cards, handlePayment }: PaymentOptionProps) {
  return (
    <S.Wrapper>
      <S.Body>
        <Heading bottomBorder color="black" size="small">
          Payment
        </Heading>

        <S.CardItem>
          <S.RadioGroupRoot>
            {cards &&
              cards.map((card) => (
                <S.ItemWrapper key={card.number}>
                  <label htmlFor={card.number}>
                    <img src={card.img} alt={card.flag} />
                    {card.number}
                  </label>

                  <S.RadioGroupItem value={card.number}>
                    <S.RadioGroupIndicator />
                  </S.RadioGroupItem>
                </S.ItemWrapper>
              ))}
          </S.RadioGroupRoot>

          <S.AddCard role="button">
            <Plus size={14} /> Add a new credit card
          </S.AddCard>
        </S.CardItem>
      </S.Body>

      <S.Footer>
        <Button as="a" fullWidth minimal>
          Continue shopping
        </Button>

        <Button fullWidth icon={<ShoppingCart />} onClick={handlePayment}>
          Buy now
        </Button>
      </S.Footer>
    </S.Wrapper>
  )
}
