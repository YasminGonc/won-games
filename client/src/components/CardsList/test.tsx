import { renderWithTheme } from '../../utils/tests/helpers'
import { screen } from '@testing-library/react'

import { CardsList } from '.'
import { paymentOptionsMock } from '../PaymentOptions/mock'

describe('<CardsList />', () => {
  it('should render 4 columns topics', () => {
    const { container } = renderWithTheme(
      <CardsList cards={paymentOptionsMock} />
    )

    expect(
      screen.getByRole('heading', { name: /my cards/i })
    ).toBeInTheDocument()
    expect(screen.getByText(/4325/)).toBeInTheDocument()
    expect(screen.getByText(/4326/)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
