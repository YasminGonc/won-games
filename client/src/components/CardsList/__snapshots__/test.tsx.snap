// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardsList /> should render 4 columns topics 1`] = `
.c1 {
  color: #030517;
  font-weight: 600;
  position: relative;
  margin-bottom: 3.2rem;
  font-size: 1.6rem;
}

.c1::after {
  content: '';
  position: absolute;
  left: 0;
  bottom: -1rem;
  width: 5rem;
  border-bottom: 0.5rem solid #F231A5;
}

.c1::after {
  width: 3rem;
}

.c0 {
  background-color: #FAFAFA;
  padding: 2.4rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.8rem;
}

.c3 {
  background-color: #EAEAEA;
  padding: 0.8rem 1.6rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 1.6rem;
}

<div
  class="c0"
>
  <h2
    class="c1"
    color="black"
  >
    My cards
  </h2>
  <div
    class="c2"
  >
    <div
      class="c3"
    >
      <img
        alt="visa"
        src="/img/visa.png"
      />
      **** **** **** 4325
    </div>
    <div
      class="c3"
    >
      <img
        alt="mastercard"
        src="/img/mastercard.png"
      />
      **** **** **** 4326
    </div>
  </div>
</div>
`;
